components/tcf/secondLayer/src/index.js

```diff
@@ -54,6 +54,21 @@ export default function TcfSecondLayer({
           object: {},
           value: true
         })
+        userConsent.vendor.legitimateInterests = {
+          ...userConsent.vendor.consents
```

Comment:
Shouldn't be also included the ADEVINTA_COLLECTED_CONSENTS purposes validation? Now it includes all available purposes, but maybe some day new purposes are added and rules are not changed.

Reply:
maybe, it needs a big refactor IMO. If it's fine to you, it's better to merge as it is now, and refactor all the state manipulation in another PR